spec/requests/tickets_spec.rb

```diff
@@ -0,0 +1,201 @@
+    {
+      contact_id: contact2.id
+    }
+  end
+
+  let(:update_additional_attributes) do
+    {
+      contact_id: contact2.id,
+      ticket_id: 'new_id'
+    }
+  end
+
+  let(:invalid_update_attributes) do
+    {
+      unexpected_property: 'value'
+    }
+  end
+
+  # Tests
+  describe 'GET /index' do
+    it 'renders a successful response' do
+      Ticket.create! base_attributes
+      get tickets_url, as: :json
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET /show' do
+    it 'renders a successful response' do
+      ticket = Ticket.create! base_attributes
+      get ticket_url(ticket), as: :json
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'POST /create' do
+    context 'with valid parameters' do
+      it 'creates a new Ticket' do
+        expect do
+          post tickets_url,
+               params: create_attributes, as: :json
+        end.to change(Ticket, :count).by(1)
+      end
+
+      it 'creates ten new Tickets' do
+        expect do
+          post tickets_url,
+               params: create_ten_attributes, as: :json
+        end.to change(Ticket, :count).by(10)
+      end
+
+      it 'renders a JSON response with the new ticket' do
+        post tickets_url,
+             params: create_attributes, as: :json
+        expect(response).to have_http_status(:created)
+        expect(response.content_type).to match(a_string_including('application/json'))
+      end
+    end
+
+    context 'with invalid parameters' do
+      it 'does not create a new Ticket without a valid participating seller' do
+        expect do
+          post tickets_url,
+               params: invalid_participating_seller_attributes, as: :json
+        end.to change(Ticket, :count).by(0)
+      end
+
+      it 'renders a JSON response with errors for the new ticket without a valid participating seller' do
+        post tickets_url,
+             params: invalid_participating_seller_attributes, as: :json
+        expect(response).to have_http_status(:not_found)
+        expect(response.content_type).to match(a_string_including('application/json'))
+      end
+
+      it 'does not create a new Ticket without a valid number of tickets' do
+        expect do
+          post tickets_url,
+               params: invalid_number_attributes, as: :json
+        end.to change(Ticket, :count).by(0)
+      end
+
+      it 'renders a JSON response with errors for the new ticket without a valid number of tickets' do
+        post tickets_url,
+             params: invalid_number_attributes, as: :json
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response.content_type).to match(a_string_including('application/json'))
+      end
+    end
+  end
+
+  describe 'PATCH /update' do
+    context 'with valid parameters' do
+      it 'updates the requested ticket with a valid contact' do
+        ticket = Ticket.create! base_attributes
+        patch ticket_url(ticket),
```

Comment:
We've never used patch before — does put also work?

Reply:
They're the same in the rails world apparently, but I'll switch these to put to be more consistent